uc.c

```diff
@@ -686,6 +686,27 @@ static void clear_deleted_hooks(uc_engine *uc)
     list_clear(&uc->hooks_to_del);
 }
 
+// set a data ptr (for use in bindings)
+UNICORN_EXPORT
+uc_err uc_emu_set_data_ptr(uc_engine *uc, void *data)
```

Comment:
Do we really need this API? I mean, why don't you save this data in the rust bindings?

Reply:
I need to know the pointer to the object somehow in the callbacks, if I store it earlier it may have been moved.
Right now it's solved in a way where the callback gets a different type than the surrounding object which also seems wrong. I can think about it again